CHANGELOG.md

```diff
@@ -1,250 +1,310 @@
+- The new `Routes` component replaces `renderRoutes()`
+- The new `MagentoRoute` component replaces `MagentoRouteHandler`
+- The new `useMagentoRoute()` talon replaces `MagentoRouteHandler`
 
-The Venia project is now split into separate `venia-ui` and `venia-concept` packages.
-This paves the way for better extensibility and customizability when it comes to storefront creation.
+### State management refactors
 
-The `venia-ui` package contains the templates and components used to create the Venia storefront.
-This package is meant to be used as a component library for custom storefront projects.
+This release refactors the way PWA Studio handles state management in Venia.
+Instead of using Redux directly, Peregrine now provides a set of context providers and hooks to interact with the different slices of state in the global data store.
 
-The `venia-concept` package contains the project scripts and configurations used to build and run the actual Venia storefront project.
+For more information, read [State management](https://pwastudio.io/technologies/basic-concepts/state-management/)
 
-This change may also require you to update import paths in your project.
+### REST to GraphQL migration
 
-See PR [#1499][]
+With the increase in GraphQL coverage in the latest Magento release (2.3.4), PWA Studio continues to refactor out REST usage in favor of GraphQL.
 
-#### New build tools
+Various usage of REST have been converted to GraphQL.
+These changes include the various cart interactions, sign-in/sign-out, and fetching country data.
 
-This release contains new tools for building, bundling, and configuring storefront projects.
+### Performance improvements
 
-**`buildpack` CLI command** - A command line toolkit with subcommands for:
+A lot of work has been done in this release to improve the performance provided by PWA Studio tools and libraries.
 
--   Generating SSL certificates
--   Creating a `.env` file
--   Loading and validating a project's `.env` file.
+#### Service worker improvements
 
-See PR [#1498][]
+Service worker changes in this release provides smarter use of the cache and when to invalidate stale data.
+Other improvements include more optimized bundles/images and route handling.
 
-**`configureWebpack()`** - A function that provides an opinionated configuration for Webpack.
-This replaces a large portion of the `webpack.config.js` file in the Venia project in favor of a configurable utility that can be used in other storefront projects.
+#### Optimized images
 
-**`babel-preset-peregrine`** - A new package that defines a preset for Babel.
-This preset is required for storefronts using Peregrine and Venia-UI components.
-See PR [#1404][].
+Handling images on the storefront has been improved in this release.
+Storefronts are now able to load the optimal image for a given viewport.
+
+New features such as pre-fetching and lazy loading also boosts page load performance.
+
+### Refactoring classes to functional components
```

Comment:
Do we have any link-able docs relating to our refactor? Maybe a Jimmy-created Gist or something?